File: lib/common/coordinate.dart
import '../model/amap_base_model.dart';

class Coordinate extends AMapBaseModel {
  final double latitude;
  final double longitude;

  Coordinate(this.latitude, this.longitude);

  @override
  Map<String, dynamic> toJson() {
    return {
      'latitude': latitude,
      'longitude': longitude,
    };
  }
}

File: android/src/main/java/amap/com/example/flutter_amap_plugin/Map/Coordinate.java
package amap.com.example.flutter_amap_plugin.Map;


import com.amap.api.maps.model.CameraPosition;
import com.amap.api.maps.model.LatLng;

public class Coordinate {
    public float latitude;
    public float longitude;

    CameraPosition toCameraPosition() {
        return new CameraPosition(new LatLng(latitude,longitude), 10, 0, 0);
    }

    LatLng toLatLng() {
        return new LatLng(latitude, longitude);
    }
}
